### PR TITLE
Sweet alert - Subtract quantity after order - Check quantity before o…

### PR DESCRIPTION
**Pull request by hduong1207uet**
## Related Tickets:
- [#35970](https://edu-redmine.sun-asterisk.vn/issues/35970)

## WHAT this PR does?
- Update quantity in stock after acacept user order
- Sweet alert in backend
- Check product quantity in cart before user order

## Notes Impacted Areas:
- N/A.

## Attachments:
1. Subtract quantity after accept order
- subtractQuantity

2. Check quantity
- image

- image

### CODE REVIEW

**Comment by hduong1207uet** [comment]
ready

File: resources/views/client/cart/index.blade.php [hunk 1]
```diff
@@ -35,6 +35,7 @@
                         <td data-th="Price">${{ $details['price'] }}</td>
                         <td data-th="quantity">
                             <input type="number" name="quantity" initial-quantity="{{ $details['quantity'] }}" id="cart_quantity_{{ $id }}" data-id="{{ $id }}" value="{{ $details['quantity'] }}" class="cart_quantity form-control quantity" min=1 max="{{ $productQuantity[$id] }}"/>
+                            <p>{!! __('in_stock') .': <b class="text-primary">' .$productQuantity[$id] .'</b>' !!}</p>
```
Comment (ghost) [inline resources/views/client/cart/index.blade.php]:
```suggestion
                            <p>{!! __('in_stock') . ': <b class="text-primary">' . $productQuantity[$id] . '</b>' !!}</p>
```

**Comment by hduong1207uet** [comment]
ready

File: app/Http/Controllers/Admin/OrderController.php [hunk 2]
```diff
@@ -76,7 +77,20 @@ public function approve(Request $request) {
         $order = Order::findOrFail($request->orderId);
         $order->update([
             'status' => config('app.approved_order_status'),
-        ]);        
+        ]);
+        
+        $orderDetails = $order->orderDetails()->get();
```
Comment (huyenttm-0901) [inline app/Http/Controllers/Admin/OrderController.php]:
```suggestion
        $orderDetails = $order->orderDetails;
```

File: app/Http/Controllers/Admin/OrderController.php [hunk 3]
```diff
@@ -76,7 +77,20 @@ public function approve(Request $request) {
...
+        foreach ($orderDetails as $orderDetail) {
+            $productOrder = Product::findOrFail($orderDetail->product_id);
```
Comment (huyenttm-0901) [inline app/Http/Controllers/Admin/OrderController.php]: n+1 query

File: app/Http/Controllers/HomeController.php [hunk 4]
```diff
@@ -150,11 +155,23 @@ public function viewProductDetail($id)
      */
     public function buyProducts()
     {
-        if (session('cart') == NULL) {
-            return redirect(route('cart'))->with('error', __('order_failed_because_your_cart_is_empty'));
+        $productIds = array_keys(session('cart'));        
+        $quantitysInStock = Product::whereIn('id', $productIds)->get()->pluck('quantity_in_stock', 'id')->toArray();
+
+        if (session('cart') != NULL) {
+            foreach(session('cart') as $id => $detail) {
```
Comment (huyenttm-0901) [inline app/Http/Controllers/HomeController.php]:
```suggestion
            foreach (session('cart') as $id => $detail) {
```

File: resources/lang/en.json [hunk 5]
```diff
@@ -164,5 +164,7 @@
     "pending" : "Pending",
     "approved" : "Approved",
     "denied" : "Denied",
-    "product_left" : "products left"  
+    "product_left" : "products left",
+    "order_failed_because_product_quantity_in_your_cart_exceed_quantity_in_stock" : "Order failed because product quantity in your cart exceed quantity in stock.",
```
Comment (huyenttm-0901) [inline resources/lang/en.json]: key em để ngắn thôi

**Comment by hduong1207uet** [comment]
ready

File: app/Http/Controllers/Admin/OrderController.php [hunk 6]
```diff
@@ -60,35 +61,48 @@ public function destroy($id)
...
+        ]);
+
+        $orderDetails = $order->load('orderDetails')->toArray();
+        foreach ($orderDetails['order_details'] as $orderDetail) {
+            $productOrder = Product::findOrFail($orderDetail['product_id']);
+            $quantityLeft = $productOrder->quantity_in_stock - $orderDetail['quantities'];
+            if ($quantityLeft >= 0) {
+                $productOrder->update([
+                    'quantity_in_stock' => $quantityLeft,
+                ]);
```
Comment (ghost) [inline app/Http/Controllers/Admin/OrderController.php]: n+1

**Comment by hduong1207uet** [comment]
ready

**Review by ghost — APPROVED** [review]

**Review by huyenttm-0901 — APPROVED** [review]

**Pull request merged**